src/test/java/core/basesyntax/dao/animal/AnimalDaoImplTest.java

```diff
@@ -1,5 +1,61 @@
 package core.basesyntax.dao.animal;
 
-public class AnimalDaoImplTest {
+import core.basesyntax.dao.AbstractTest;
+import core.basesyntax.model.zoo.Animal;
+import core.basesyntax.model.zoo.Cat;
+import core.basesyntax.model.zoo.Dog;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
 
+public class AnimalDaoImplTest extends AbstractTest {
+    private AnimalDao dao;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{
+                Animal.class,
+                Cat.class,
+                Dog.class
+        };
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        dao = new AnimalDaoImpl(getSessionFactory());
+    }
+
+    @Test
+    public void createCat_Ok() {
+        Cat cat = new Cat();
+        cat.setAge(4);
+        cat.setName("barsik");
+        cat.setColor("gray");
+        cat.setNumberOfLives(7);
+        Cat actual = (Cat) dao.save(cat);
+        Assert.assertNotNull(actual);
+        Assert.assertNotNull(actual.getId());
+        Assert.assertEquals(1L, actual.getId().longValue());
+    }
+
+    @Test
+    public void findByName_Ok() {
```

Comment:
Not bad) Let's add more test cases. For example, when we don't have any dogs with a suitable name (e.g., returned list should be empty)

Reply:
added